client/lib/deterministic-stringify/index.js

```diff
@@ -1,32 +1,39 @@
-function deterministStringify( source ) {
-	var namespace = [],
-		keys;
+export function deterministicStringify( source ) {
+	if ( source === null ) {
+		return 'null';
+	}
+
+	// Handle primitive data types:
+	// boolean, number, string, undefined
 	if ( typeof source !== 'object' ) {
 		if ( typeof source === 'undefined' ) {
 			return 'undefined';
 		}
+
 		if ( typeof source === 'string' ) {
-			return "'" + source.toString() + "'";
+			return `'${ source.toString() }'`;
 		}
+
 		return source.toString();
 	}
+
+	// arrays are objects too so we have to
+	// manually ask if this is _also_ an array
 	if ( Array.isArray( source ) ) {
-		source = source.sort();
-		source.forEach( function( item ) {
-			namespace.push( deterministStringify( item ) );
-		} );
-		return namespace.join( ',' );
-	}
-	if ( source === null ) {
-		return 'null';
+		return source
+			.slice() // sort mutates the original array!
+			.sort()
+			.map( deterministicStringify )
+			.join( ',' );
 	}
 
-	keys = Object.keys( source );
-	keys.sort();
-	keys.forEach( function( key ) {
-		namespace.push( key + '=' + deterministStringify( source[ key ] ) );
-	} );
-	return namespace.join( '&' );
+	// else return stringified object with
+	// keys in alphabetical order
+	return Object
+		.keys( source )
+		.sort()
+		.map( key => `${ key }=${ deterministicStringify( source[ key ] ) }` )
+		.join( '&' );
 }
 
-module.exports = deterministStringify;
+export default deterministicStringify;
```

Comment:
Since we export both a default and `deterministicStringify` all old requires need to be updated, including the matching test.

eg `require( 'lib/deterministic-stringify' ).default` or update that to an import